File: tests/LanAudioRelay.Tests/OpusCodecTests.cs
using LanAudioRelay.Core.Audio;

namespace LanAudioRelay.Tests;

public sealed class OpusCodecTests
{
    [Fact]
    public void Opus_encoder_decoder_round_trips_a_stereo_frame()
    {
        var encoder = new OpusFrameEncoder();
        var decoder = new OpusFrameDecoder();
        var pcm = new short[AudioConstants.PcmSamplesPerFrame];

        for (var index = 0; index < AudioConstants.SamplesPerFrame; index++)
        {
            var sample = (short)(Math.Sin(index / 12.0) * 12_000);
            pcm[index * 2] = sample;
            pcm[index * 2 + 1] = sample;
        }

        var payload = encoder.Encode(pcm);
        var decoded = decoder.Decode(payload);

        Assert.NotEmpty(payload);
        Assert.Equal(AudioConstants.PcmSamplesPerFrame, decoded.Length);
        Assert.Contains(decoded, sample => sample != 0);
    }
}

File: mac/LanAudioRelayMac/Tests/LanAudioRelayMacCoreTests/OpusCodecTests.swift
import XCTest
@testable import LanAudioRelayMacCore

final class OpusCodecTests: XCTestCase {
    func testOpusEncoderDecoderRoundTripsStereoFrame() throws {
        let encoder = try OpusFrameEncoder()
        let decoder = try OpusFrameDecoder()
        var pcm = [Int16](repeating: 0, count: AudioConstants.pcmSamplesPerFrame)

        for frame in 0..<AudioConstants.samplesPerFrame {
            let sample = Int16(Double.sin(Double(frame) / 12.0) * 12_000)
            pcm[frame * 2] = sample
            pcm[frame * 2 + 1] = sample
        }

        let payload = try encoder.encode(pcm)
        let decoded = try decoder.decode(payload)

        XCTAssertFalse(payload.isEmpty)
        XCTAssertEqual(decoded.count, AudioConstants.pcmSamplesPerFrame)
        XCTAssertTrue(decoded.contains { $0 != 0 })
    }
}
